text/0064-publishing-protobufs.md

```diff
@@ -0,0 +1,198 @@
+
+
+For most clients, the Buf Schema Registry is the recommended way to use the
+MobileCoin Foundation's Protobufs.
+
+## Rust Crates
+
+The MobileCoin Foundation Protobuf defintions are available via
+[Rust](https://www.rust-lang.org/) crates.  The crates use
+[prost](https://docs.rs/prost/latest/prost/) generated messages. Both
+[tonic](https://docs.rs/tonic/latest/tonic/) and
+[grpcio](https://docs.rs/grpcio/latest/grpcio/) are supported for the services.
+
+
+These crates can be found on
+<https://crates.io> by searching for one of the following:
+
+- [mc-*-grpcio](https://crates.io/search?q=mc-*-grpcio) for rust
+  [grpcio](https://docs.rs/grpcio/latest/grpcio/) services.
+- [mc-*-messages](https://crates.io/search?q=mc-*-messages) for
+  [prost](https://docs.rs/prost/latest/prost/) messages.
+- [mc-*-tonic](https://crates.io/search?q=mc-*-tonic) for
+  [tonic](https://docs.rs/tonic/latest/tonic/) services.
+
+Building the crates requires the client to have the Protocol Buffer Compiler,
+[protoc](https://grpc.io/docs/protoc-installation/), installed.
+
+For rust projects, the rust crates are the recommended way of using the MobileCoin Foundatin Protobuf defintions.
+
+## The Protobuf Git Repo
+
+Clients can directly access the Protobuf definitions via the git repo,
+<https://github.com/mobilecoinfoundation/protobufs>. 
+
+The other two methods should be prefered over direct access to the git repo, but
+there are times when it is necessary to use the repo directly.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The MobileCoin Foundation Protobuf defintions are located at
+<https://github.com/mobilecoinfoundation/protobufs>. When a series of git
+changes are deemed ready to release, the new versions will be published to
+<https://buf.build/mobilecoinfoundation> and crates published to <https://crates.io>. 
+
+The Protobuf messages are placed in separate files from the Protobuf services.
+This makes it easier for one to build just the message types without needing to
+worry about the service implementation.
+
+Each Protobuf API will be placed in it's own directory with the
+following layout:
+
+```console
+mobilecoinfoundation
+└── <name of api>
+    ├── grpcio
+    │   ├── build.rs
+    │   ├── Cargo.toml
+    │   ├── protos
+    │   │   └── services.proto (Symlink to <version>/services.proto)
+    │   ├── README.md
+    │   └── src
+    │       └── lib.rs
+    ├── messages
+    │   ├── build.rs
+    │   ├── Cargo.toml
+    │   ├── protos
+    │   │   └── messages.proto (Symlink to <version>/messages.proto)
+    │   ├── README.md
+    │   └── src
+    │       └── lib.rs
+    ├── tonic
+    │   ├── build.rs
+    │   ├── Cargo.toml
+    │   ├── protos
+    │   │   └── services.proto (Symlink to <version>/services.proto)
+    │   ├── README.md
+    │   └── src
+    │       └── lib.rs
+    └── <version> (of the form `v1`, `v2`, etc.)
+        ├── messages.proto
+        └── services.proto
```

Comment:
Currently the attestation messages are in `attestation.proto` but in writing this up it seemed odd to have a `services.proto` and then a `whatever.proto`